src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -737,7 +753,7 @@ private void visitClassOrInterface(String name, FunctionType ftype) {
       // Class extends another class
       if (getSuperType(ftype) != null) {
         emit("extends");
-        visitType(getSuperType(ftype));
+        visitType(getSuperType(ftype), emitInstance);
```

Comment:
Good catch, tried doing it and found a bug with templatized classes.
At that point we have two cases visited in visitType, so it is not so simple as copying them over. So I am leaving a comment for now.